# Issue 12871: Conversion for RR[x][y] stricter than for QQ[x][y]

archive/issues_012699.json:
```json
{
    "assignees": [
        "https://github.com/aghitza"
    ],
    "body": "<div id=\"comment:0\"></div>\n\nThe following works:\n\n```\nsage: R.<x,y> = QQ[]\nsage: U = R.univariate_ring(x)\nsage: U([y])\ny\n```\nbut over ``RR`` it does not:\n\n```\n\nsage: R.<x,y> = RR[]\nsage: U = R.univariate_ring(x)\nsage: U([y])\n---------------------------------------------------------------------------\nValueError                                Traceback (most recent call last)\n\n/home/vbraun/<ipython console> in <module>()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/parent.so in sage.structure.parent.Parent.__call__ (sage/structure/parent.c:7940)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3344)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3247)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_ring.pyc in _element_constructor_(self, x, check, is_gen, construct, **kwds)\n    364         C = self._polynomial_class\n    365         if isinstance(x, list):\n--> 366             return C(self, x, check=check, is_gen=False,construct=construct)\n    367         if isinstance(x, Element):\n    368             P = x.parent()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_element.so in sage.rings.polynomial.polynomial_element.Polynomial_generic_dense.__init__ (sage/rings/polynomial/polynomial_element.c:38456)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/parent.so in sage.structure.parent.Parent.__call__ (sage/structure/parent.c:7940)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.NamedConvertMap._call_ (sage/structure/coerce_maps.c:4298)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/multi_polynomial.so in sage.rings.polynomial.multi_polynomial.MPolynomial._polynomial_ (sage/rings/polynomial/multi_polynomial.c:3371)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/multi_polynomial.so in sage.rings.polynomial.multi_polynomial.MPolynomial.polynomial (sage/rings/polynomial/multi_polynomial.c:5417)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/parent.so in sage.structure.parent.Parent.__call__ (sage/structure/parent.c:7940)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3344)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3247)()\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_ring.pyc in _element_constructor_(self, x, check, is_gen, construct, **kwds)\n    416             if x.type() != 't_POL':\n    417                 x = x.Polrev()\n--> 418         return C(self, x, check, is_gen, construct=construct, **kwds)\n    419 \n    420     def is_integral_domain(self, proof = True):\n\n/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_real_mpfr_dense.so in sage.rings.polynomial.polynomial_real_mpfr_dense.PolynomialRealDense.__init__ (sage/rings/polynomial/polynomial_real_mpfr_dense.c:3449)()\n\nValueError: max() arg is an empty sequence\n```\nThis is the reason for the behavior observed in http://ask.sagemath.org/question/1359/strange-error-with-multivariate-polynomials\n\nComponent: **algebra**\n\n_Issue created by migration from https://trac.sagemath.org/ticket/12871_\n\n",
    "closed_at": "2015-09-30T21:51:50Z",
    "created_at": "2012-04-24T04:11:38Z",
    "labels": [
        "https://github.com/sagemath/sage/labels/bug",
        "https://github.com/sagemath/sage/labels/c%3A%20algebra",
        "https://github.com/sagemath/sage/labels/duplicate"
    ],
    "reactions": [],
    "repository": "https://github.com/sagemath/sage",
    "title": "Conversion for RR[x][y] stricter than for QQ[x][y]",
    "type": "issue",
    "updated_at": "2015-09-30T21:51:50Z",
    "url": "https://github.com/sagemath/sage/issues/12871",
    "user": "https://github.com/vbraun"
}
```
<div id="comment:0"></div>

The following works:

```
sage: R.<x,y> = QQ[]
sage: U = R.univariate_ring(x)
sage: U([y])
y
```
but over ``RR`` it does not:

```

sage: R.<x,y> = RR[]
sage: U = R.univariate_ring(x)
sage: U([y])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)

/home/vbraun/<ipython console> in <module>()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/parent.so in sage.structure.parent.Parent.__call__ (sage/structure/parent.c:7940)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3344)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3247)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_ring.pyc in _element_constructor_(self, x, check, is_gen, construct, **kwds)
    364         C = self._polynomial_class
    365         if isinstance(x, list):
--> 366             return C(self, x, check=check, is_gen=False,construct=construct)
    367         if isinstance(x, Element):
    368             P = x.parent()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_element.so in sage.rings.polynomial.polynomial_element.Polynomial_generic_dense.__init__ (sage/rings/polynomial/polynomial_element.c:38456)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/parent.so in sage.structure.parent.Parent.__call__ (sage/structure/parent.c:7940)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.NamedConvertMap._call_ (sage/structure/coerce_maps.c:4298)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/multi_polynomial.so in sage.rings.polynomial.multi_polynomial.MPolynomial._polynomial_ (sage/rings/polynomial/multi_polynomial.c:3371)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/multi_polynomial.so in sage.rings.polynomial.multi_polynomial.MPolynomial.polynomial (sage/rings/polynomial/multi_polynomial.c:5417)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/parent.so in sage.structure.parent.Parent.__call__ (sage/structure/parent.c:7940)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3344)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/structure/coerce_maps.so in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (sage/structure/coerce_maps.c:3247)()

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_ring.pyc in _element_constructor_(self, x, check, is_gen, construct, **kwds)
    416             if x.type() != 't_POL':
    417                 x = x.Polrev()
--> 418         return C(self, x, check, is_gen, construct=construct, **kwds)
    419 
    420     def is_integral_domain(self, proof = True):

/home/vbraun/opt/sage-5.0.beta13/local/lib/python2.7/site-packages/sage/rings/polynomial/polynomial_real_mpfr_dense.so in sage.rings.polynomial.polynomial_real_mpfr_dense.PolynomialRealDense.__init__ (sage/rings/polynomial/polynomial_real_mpfr_dense.c:3449)()

ValueError: max() arg is an empty sequence
```
This is the reason for the behavior observed in http://ask.sagemath.org/question/1359/strange-error-with-multivariate-polynomials

Component: **algebra**

_Issue created by migration from https://trac.sagemath.org/ticket/12871_





---

archive/issue_events_176014.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2012-04-24T04:11:38Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-5.11",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176014"
}
```



---

archive/issue_events_176015.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2012-04-24T04:11:38Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/c%3A%20algebra",
    "label_color": "0000ff",
    "label_name": "c: algebra",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176015"
}
```



---

archive/issue_events_176016.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2012-04-24T04:11:38Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20major%20/%203",
    "label_color": "ffbb00",
    "label_name": "p: major / 3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176016"
}
```



---

archive/issue_events_176017.json:
```json
{
    "actor": "https://github.com/aghitza",
    "created_at": "2012-04-24T04:11:38Z",
    "event": "assigned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "subject": "https://github.com/vbraun",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176017"
}
```



---

archive/issue_events_176018.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2012-04-24T04:12:00Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/bug",
    "label_color": "d73a4a",
    "label_name": "bug",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176018"
}
```



---

archive/issue_events_176019.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-08-13T15:35:53Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-5.11",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176019"
}
```



---

archive/issue_events_176020.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-08-13T15:35:53Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-5.12",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176020"
}
```



---

archive/issue_events_176021.json:
```json
{
    "actor": "https://github.com/sagetrac-vbraun-spam",
    "created_at": "2014-01-30T21:20:52Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.1",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176021"
}
```



---

archive/issue_events_176022.json:
```json
{
    "actor": "https://github.com/sagetrac-vbraun-spam",
    "created_at": "2014-01-30T21:20:52Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.2",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176022"
}
```



---

archive/issue_events_176023.json:
```json
{
    "actor": "https://github.com/sagetrac-vbraun-spam",
    "created_at": "2014-05-06T15:20:58Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.2",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176023"
}
```



---

archive/issue_events_176024.json:
```json
{
    "actor": "https://github.com/sagetrac-vbraun-spam",
    "created_at": "2014-05-06T15:20:58Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.3",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176024"
}
```



---

archive/issue_events_176025.json:
```json
{
    "actor": "https://github.com/sagetrac-vbraun-spam",
    "created_at": "2014-08-10T16:51:03Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.3",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176025"
}
```



---

archive/issue_events_176026.json:
```json
{
    "actor": "https://github.com/sagetrac-vbraun-spam",
    "created_at": "2014-08-10T16:51:03Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.4",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176026"
}
```



---

archive/issue_events_176027.json:
```json
{
    "actor": "https://github.com/bgrenet",
    "created_at": "2015-09-25T14:57:56Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176027"
}
```



---

archive/issue_events_176028.json:
```json
{
    "actor": "https://github.com/bgrenet",
    "created_at": "2015-09-25T15:00:30Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "milestone_number": null,
    "milestone_title": "sage-6.4",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176028"
}
```



---

archive/issue_events_176029.json:
```json
{
    "actor": "https://github.com/bgrenet",
    "created_at": "2015-09-25T15:00:30Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176029"
}
```



---

archive/issue_events_176030.json:
```json
{
    "actor": "https://github.com/bgrenet",
    "created_at": "2015-09-25T15:00:30Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176030"
}
```



---

archive/issue_comments_145191.json:
```json
{
    "body": "<div id=\"comment:7\" align=\"right\">comment:7</div>\n\nIt was not noticed that #17190 is a duplicate of this ticket. I fixed the problem in #17190:\n\n```python\nsage: R.<x,y> = RR[]\nsage: U = R.univariate_ring(x)\nsage: U([y])\ny\n```",
    "created_at": "2015-09-25T15:00:30Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/12871#issuecomment-145191",
    "user": "https://github.com/bgrenet"
}
```

<div id="comment:7" align="right">comment:7</div>

It was not noticed that #17190 is a duplicate of this ticket. I fixed the problem in #17190:

```python
sage: R.<x,y> = RR[]
sage: U = R.univariate_ring(x)
sage: U([y])
y
```



---

archive/issue_events_176031.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2015-09-30T21:51:50Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176031"
}
```



---

archive/issue_events_176032.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2015-09-30T21:51:50Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176032"
}
```



---

archive/issue_events_176033.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2015-09-30T21:51:50Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20major%20/%203",
    "label_color": "ffbb00",
    "label_name": "p: major / 3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176033"
}
```



---

archive/issue_events_176034.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2015-09-30T21:51:50Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/12871",
    "label": "https://github.com/sagemath/sage/labels/duplicate",
    "label_color": "c6c6c6",
    "label_name": "duplicate",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/12871#event-176034"
}
```
